test/components/birth-date-picker.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import { ClientRenderer, expect, simulate, sinon, waitFor } from "test-drive-react";
+import { BirthDatePicker, dateFromYearMonthDay } from "../../src";
+import { BirthDatePickerDemo } from "../../demo/birth-date-picker-demo";
+
+function updateInput(input: HTMLInputElement, value: string) {
+    input.focus();
+    input.value = value;
+    simulate.change(input);
+}
+
+describe("<BirthDatePicker />", () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
+
+    it("Displays the provided date and allows changing it", async function() {
```

Comment:
I suggest that if you enrich the demo (see above, show some side effect) you can test here the effects on the demo itself, not component internals. In other words, I'd make the component test less unit-testey.